### PR TITLE
Add NuGetTest.sln to build_test_all.cmd

### PR DESCRIPTION
**Pull request by jlaanstra**
Build the NuGetTest solution as part of build_test_all.cmd.

TODO:
- [x] Run a build to make sure it works.

### CODE REVIEW

File: build_test_all.cmd [hunk 1]
```diff
@@ -8,9 +8,6 @@ if "%target_platform%"=="" set target_platform=x64
 if "%target_configuration%"=="" set target_configuration=Release
 if "%target_version%"=="" set target_version=1.2.3.4
 
-call nuget restore cppwinrt.sln
```
Comment (kennykerr) [inline build_test_all.cmd]: This is pretty important. Why did you remove it?

File: build_test_all.cmd [hunk 2]
```diff
@@ -19,13 +16,13 @@ if "%target_platform%"=="arm64" goto :eof
...
-call msbuild /m /p:Configuration=%target_configuration%,Platform=%target_platform%,CppWinRTBuildVersion=%target_version% cppwinrt.sln /t:test\test_win7
 call msbuild /m /p:Configuration=%target_configuration%,Platform=%target_platform%,CppWinRTBuildVersion=%target_version% cppwinrt.sln /t:test\test_fast
 call msbuild /m /p:Configuration=%target_configuration%,Platform=%target_platform%,CppWinRTBuildVersion=%target_version% cppwinrt.sln /t:test\test_slow
 call msbuild /m /p:Configuration=%target_configuration%,Platform=%target_platform%,CppWinRTBuildVersion=%target_version% cppwinrt.sln /t:test\test_module_lock_custom
 call msbuild /m /p:Configuration=%target_configuration%,Platform=%target_platform%,CppWinRTBuildVersion=%target_version% cppwinrt.sln /t:test\test_module_lock_none
 call msbuild /m /p:Configuration=%target_configuration%,Platform=%target_platform%,CppWinRTBuildVersion=%target_version% cppwinrt.sln /t:test\old_tests\test_old
 
-call run_tests.cmd %target_platform% %target_configuration%
+rem call run_tests.cmd %target_platform% %target_configuration% %target_version%
```
Comment (kennykerr) [inline build_test_all.cmd]: This causes catch2 to emit a junit report. While I'd love to do this for build reporting, it doesn't appear to work reliably. Only turn this on if you're sure it works reliably now.

File: build_test_all.cmd [hunk 3]
```diff
@@ -19,13 +16,13 @@ if "%target_platform%"=="arm64" goto :eof
 call msbuild /m /p:Configuration=%target_configuration%,Platform=%target_platform%,CppWinRTBuildVersion=%target_version% cppwinrt.sln /t:cppwinrt
 _build\%target_platform%\%target_configuration%\cppwinrt.exe -in local -out _build\%target_platform%\%target_configuration% -verbose
 call msbuild /p:Configuration=%target_configuration%,Platform=%target_platform%,CppWinRTBuildVersion=%target_version% natvis\cppwinrtvisualizer.sln
+call msbuild /p:Configuration=%target_configuration%,Platform=%target_platform%,CppWinRTBuildVersion=%target_version% test\nuget\NugetTest.sln
 
 call msbuild /m /p:Configuration=%target_configuration%,Platform=%target_platform%,CppWinRTBuildVersion=%target_version% cppwinrt.sln /t:test\test
-call msbuild /m /p:Configuration=%target_configuration%,Platform=%target_platform%,CppWinRTBuildVersion=%target_version% cppwinrt.sln /t:test\test_win7
```
Comment (kennykerr) [inline build_test_all.cmd]: It looks like you didn't merge with main before creating this PR. This should not be removed.

File: build_test_all.cmd (hunk 1 above)
Comment (jlaanstra) [reply build_test_all.cmd]: Not intentional. Let me rebase the branch.

**Comment by jlaanstra** [comment]
Should be fixed now. Can I kick off a build to verify it works?

**Comment by kennykerr** [comment]
Go for it.

File: build_test_all.cmd [hunk 4]
```diff
@@ -19,6 +19,7 @@ if "%target_platform%"=="arm64" goto :eof
 call msbuild /m /p:Configuration=%target_configuration%,Platform=%target_platform%,CppWinRTBuildVersion=%target_version% cppwinrt.sln /t:cppwinrt
 _build\%target_platform%\%target_configuration%\cppwinrt.exe -in local -out _build\%target_platform%\%target_configuration% -verbose
 call msbuild /p:Configuration=%target_configuration%,Platform=%target_platform%,CppWinRTBuildVersion=%target_version% natvis\cppwinrtvisualizer.sln
+call msbuild /p:Configuration=%target_configuration%,Platform=%target_platform%,CppWinRTBuildVersion=%target_version% test\nuget\NugetTest.sln
```
Comment (kennykerr) [inline build_test_all.cmd]: I think you still need a nuget restore for NugetTest.sln - try running "git clean fdx" and running "build_test_all" - that should work.

Comment (jlaanstra) [reply build_test_all.cmd]: NuGetTest.sln doesn't have any NuGet package dependencies so shouldn't be necessary.

Comment (kennykerr) [reply build_test_all.cmd]: Oh it grabs cppwinrt locally? Cool.

Comment (jlaanstra) [reply build_test_all.cmd]: Turns out I was wrong. The C# project has nuget dependencies.

**Comment by jlaanstra** [comment]
Closing this and creating a new PR directly from origin.
